src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl

```diff
@@ -1577,12 +1577,14 @@ KERNEL(sdpa_opt)(
                     VALUE_COMPRESSION_SCALE_TYPE comp_zp = val_scale[comp_offset + 1];
 #endif
 #endif
+                    // separate value read out of the loop for reducing register pressure
+                    #ifdef V_HEAD_SIZE_LEFTOVER && !defined(BEAM_TABLE_TYPE)
+                    const INPUT2_TYPE value_packed = (sgid < SUBGROUPS_PER_WG - 1) ? VALUE_BLOCK_READ(value_input, value_offset) : head_size_idx < V_HEAD_SIZE ? value_input[value_offset] : INPUT2_VAL_ZERO;
+                    #endif
                     unroll_for (uint i = 0; i < SUBGROUP_SIZE; i++) {
```

Comment:
This loop is supposed to process one block with size like [..., SUBGROUP_SIZE, HEAD_SIZE] which is a part of Value input [..., SEQ_LEN, HEAD_SIZE], so we have CeilDiv(SEQ_LEN, SUBGROUP_SIZE) such blocks
Whole workgroup (consisting of HEAD_SIZE work items) should read one row (of size HEAD_SIZE) from this "block" in each iteration of this loop, resulting in each work item reads one value and multiplies it with the corresponding broadcasted QK. Therefore, we can't move it outside the loop like this - doing so would mean we're reading only 1 row out of 16
Btw, similar logic should also be updated here as well: https://github.com/openvinotoolkit/openvino/blob/9caa9971ffb5e15bffde37bf9899a18fa924cace/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl#L1661

Reply:
you are right. I missed the value_offset increasing in the loop. I will look for other approach. Thanks!